File: LAB_03/motorBoat.java
package LAB_03;

import java.io.FileNotFoundException;
import LAB_03.interfaces.Flue;
import LAB_03.interfaces.Powered;;

public class motorBoat extends ShlupkaAbs implements Flue, Powered {
    private boolean engineRunning;
    private double fuelLiters;

    public motorBoat() throws FileNotFoundException {

        engineRunning = false;

        fuelLiters = 0;
    }

    public motorBoat(int countOars, String material, double length,
            double width, int countSeats, String nameFile,
            boolean engineRunning, double fuelLiters, double speedMS) throws FileNotFoundException {
        super(countOars, material, length, width, countSeats, nameFile);
        this.fuelLiters = fuelLiters;
        this.engineRunning = engineRunning;
    }

    @Override
    public void moveBoat(double distanceMetres) {
        if (fuelLiters != 0 && distanceMetres <= fuelLiters && engineRunning) {
            fuelLiters -= distanceMetres;

            fout.println("Boat moved " + distanceMetres + ". Remaining fuel: " + fuelLiters);
        }

        else if (!engineRunning) {
            fout.println("Cannot move. Engine turn off!");
        }

        else {
            fout.println("Cannot move. Not enough fuel!");
        }
    }

    @Override
    public void startEngine() {
        engineRunning = true;
        fout.println("Engine turn on!");
    }

    @Override
    public void stopEngine() {
        engineRunning = false;
        fout.println("Engine turn off!");
    }

    @Override
    public void checkFlue() {
        fout.println("Remaining fuel: " + fuelLiters);
    }

    @Override
    public void Refueling(double countFlue) {
        fuelLiters += countFlue;
        fout.println("Boat is refueling!. Remaining fuel: " + fuelLiters);
    }

    @Override
    public void boatInfo() {
        super.boatInfo();
        fout.println("Engine running: " + (engineRunning ? "Yes" : "No"));
        fout.println("Fuel liters: " + fuelLiters);
    }
}

File: LAB_09/motorBoat.py
import boat

class MotorBoat(boat.Boat):
    def __init__(self, countOars=2, material="wood", length=6, width=2,
                 countSeats=2, move=False, fuelLiters=0, engineRunning=False):
        super().__init__(countOars, material, length, width, countSeats, move)
        self.__fuelLiters = fuelLiters
        self.__engineRunning = engineRunning

    def startEngine(self):
        if not self.__engineRunning and self.__fuelLiters > 0:
            self.__engineRunning = True
            print("Engine started.")
        elif self.__engineRunning:
            print("Engine already running.")
        else:
            print("Cannot start engine — no fuel!")

    def stopEngine(self):
        if self.__engineRunning:
            self.__engineRunning = False
            print("Engine stopped.")
        else:
            print("Engine already stopped.")

    def refuel(self, liters):
        self.__fuelLiters += liters
        print(f"Boat refueled. Current fuel: {self.__fuelLiters} liters.")

    def checkFlue(self):
        print(f"Remaining flue: {self.__flueLiters} liters.")

    def boatInfo(self):
        super().boatInfo()
        print(f"Engine running: {self.__engineRunning}")
        print(f"Fuel liters: {self.__fuelLiters}")
        
